### ISSUE TITLE
Define segmentation classes with hypes file

### ISSUE THREAD

**Issue by MartinThoma** [TensorVision/TensorVision#42]
I would like to expand the hypes files to have an entry `classes`:

``` json
{
  "model": {
    "input_file": "inputs/cifar10_input.py",
    "architecture_file" : "architectures/cifar_net.py",
    "objective_file" : "objectives/softmax_classifier.py",
    "optimizer_file" : "optimizer/exp_decay.py"
  },

  "data": {
    "data_url" : "http://www.cs.toronto.edu/~kriz/cifar-10-binary.tar.gz",
    "num_examples_per_epoch_for_train" : 50000,
    "num_examples_per_epoch_for_eval" : 10000
  },

  "classes": [
      {"name": "road",
       "colors": [[255, 0, 255]],
       "output": [0, 255, 0, 127]},
      {"name": "background",
       "colors": ["default", [255, 0, 0]],
       "output": [255, 0, 0, 50]},
      {"name": "ignore",
       "colors": [[0, 0, 0, 0]]}
  ],

  "arch": {
    "crop_size" : 32,
    "num_classes" : 10,
    "image_size" : 24,
    "num_channels" : 3
  },

  "logging": {
  },

  "solver": {
      "opt": null,
      "batch_size": 128,
      "epsilon": 0.00001,
      "learning_rate": 0.1,
      "max_steps": 100000
  }
}
```

Classes is a list of dicts with keys `name` (convenience, so that one knows what the entry is about), `colors` (what colors - may be multiple - encode this class in the training / testing data), `output` (which color the class gets in the output). The string `"default"` for colors means that all colors in the segmentation masks which are not assigned will get this class. The class `ignore` is special. If it is there, it has to be last. It is just important for testing, where the errors of this class will be ignored.

I am still thinking if I should use HTML color notation (e.g. `#ff00ff` instead of `[255, 0, 255]`). The advantage is that it might be easier to use 3rd party tools / color pickers. It might also be easier to understand the format as the HTML color notation is very well-known.

**Comment by MartinThoma** [comment]
@MarvinTeichmann What do you think?

**Comment by MarvinTeichmann** [comment]
In general the concept sounds really good. But, as of now TensorVision is only actively using the 'model' section. All other section get passed individual modules (e.g. input producer). So the hypes file does not follow a strict format. Whether classes are in the hypes and how they are used will depend on the model trained.

**Comment by MartinThoma** [comment]
This was implemented in 09c6a87569ec6639aea14be7e4a767cdf654ccdd
